clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1174,6 +1189,13 @@ class SyclKernelDeclCreator
     return true;
   }
 
+  bool handleArrayType(FieldDecl *FD, QualType FieldTy) final {
+    RecordDecl *NewClass = wrapAnArray(SemaRef.getASTContext(), FieldTy, FD);
```

Comment:
Since you are decomposing and passing special classes individually, is this needed when array element type is special SYCL type?